src/PSRule.Rules.Azure/Data/Policy/Models.cs

```diff
@@ -108,6 +108,16 @@ public PolicyDefinition(string definitionId, string name, string description, JO
         /// </summary>
         public JObject Where { get; set; }
 
+        /// <summary>
+        /// The unique hash of the condition and pre-condition for the rule.
+        /// </summary>
+        public string HashKey { get; set; }
+
+        /// <summary>
+        /// The prefix for the policy rule.
+        /// </summary>
+        public string Prefix { get; set; }
```

Comment:
Currently we set name in the constructor however I think we can just remove it from the constructor and set it directly as `{prefix}.Policy.{hash}` later when we have a hash constructed instead of introducing two new properties which are for the same purpose.